src/components/table/react/TableCell.tsx

```diff
@@ -0,0 +1,200 @@
+ */
+enum Orders {
+    asc = 'asc',
+    desc = 'desc',
+}
+type Order = Orders;
+
+/**
+ * The authorized values for the `scope` prop.
+ */
+enum Scopes {
+    col = 'col',
+    row = 'row',
+}
+type Scope = Scopes;
+
+/**
+ * The authorized variants.
+ */
+enum Variants {
+    body = 'body',
+    head = 'head',
+}
+type Variant = Variants;
+
+/////////////////////////////
+
+/**
+ * Defines the props of the component.
+ */
+interface ITableCellProps extends IGenericProps {
+    /**
+     * The mdi name of the icon (thead only).
+     */
+    icon?: string;
+
+    /**
+     * Whether the column is sortable or not (thead only).
+     */
+    isSortable?: boolean;
+
+    /**
+     * The scope of the thead.
+     */
+    scope?: Scope;
+
+    /**
+     * The initial sort order (sortable thead only).
+     */
+    sortOrder?: Order;
+
+    /**
+     * The variant of the cell.
+     */
+    variant?: Variant;
+
+    /**
+     * The function to call when we click on an order button.
+     */
+    onHeaderClick?(): void;
+}
+type TableCellProps = ITableCellProps;
+
+/////////////////////////////
+
+/**
+ * Define the types of the default props.
+ */
+interface IDefaultPropsType extends Partial<TableCellProps> {
+    variant: Variant;
+}
+
+/////////////////////////////
+//                         //
+//    Public attributes    //
+//                         //
+/////////////////////////////
+
+/**
+ * The display name of the component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const COMPONENT_NAME: string = `${COMPONENT_PREFIX}TableCell`;
+
+/**
+ * The default class name and classes prefix for this component.
+ *
+ * @type {string}
+ * @constant
+ * @readonly
+ */
+const CLASSNAME: string = getRootClassName(COMPONENT_NAME, true);
+
+/**
+ * The default value of props.
+ *
+ * @type {IDefaultPropsType}
+ * @constant
+ * @readonly
+ */
+const DEFAULT_PROPS: IDefaultPropsType = {
+    onHeaderClick: undefined,
+    variant: Variants.body,
+};
+
+/////////////////////////////
+
+/**
+ * The TableCell component displays an HTML Table Header Cell.
+ *
+ * @return {React.ReactElement} The component.
+ */
+const TableCell: React.FC<TableCellProps> = ({
+    children,
+    className = '',
+    icon,
+    isSortable,
+    onHeaderClick = DEFAULT_PROPS.onHeaderClick,
+    sortOrder,
+    variant = DEFAULT_PROPS.variant,
+    ...props
+}: TableCellProps): React.ReactElement => {
+    /**
+     * Handle click on the ordered thead.
+     */
+    const handleOnHeaderClick: () => void = useCallback(() => {
+        if (isFunction(onHeaderClick)) {
+            onHeaderClick();
+        }
+    }, [onHeaderClick]);
+
+    return (
+        <Fragment>
+            {variant === Variants.head && (
+                <th
+                    className={classNames(
+                        handleBasicClasses({ prefix: CLASSNAME, isSortable }),
+                        className,
+                        `${CLASSNAME}--header`,
```

Comment:
Use handleBasicClasses

Reply:
How ?

Reply:
```
handleBasicClasses({
    awesome: 'class',
})
```

It will generate something like 'lumx-component--awesome-class'.

Reply:
Really don't think it's relevant here, it would be 
```js
handleBasicClasses({
    '-': '-header',
})
```

The goal here is to clearly show that the current element is a variant of the root one

Reply:
okay